thirdparty/github.go

```diff
@@ -32,54 +30,84 @@ const (
 	GithubAPIStatusGood  = "good"
 )
 
-type GithubUser struct {
-	Active       bool   `json:"active"`
-	DispName     string `json:"display-name"`
-	EmailAddress string `json:"email"`
-	FirstName    string `json:"first-name"`
-	LastName     string `json:"last-name"`
-	Name         string `json:"name"`
+func githubRetry(resp *github.Response, err error) (bool, error) {
+	if _, ok := err.(*github.RateLimitError); ok {
+		return false, err
+	}
+	if resp.StatusCode == http.StatusBadRequest {
+		return false, err
+	}
+	if resp.StatusCode == http.StatusNotFound {
```

Comment:
Do we actually want to retry in this case because we have seen eventual consistency problems with GitHub?

Reply:
Given that we can't distinguish between genuinely not found and github consistency problems, I ere on the side of not retrying. We will consume more github api limits if we retry on 404. We can apply special logic/logging to patch_intents (which are unaffected by these changes)